app/assets/javascripts/admin.js

```diff
@@ -24,7 +24,8 @@ var assignmentsReady = function(){
 
 }
 
-$(document).ready(assignmentsReady)
+$(document).ready(assignmentsReady);
+$(document).on('page:load', assignmentsReady);
```

Comment:
I think the concern was not about them firing at the same time, but rather firing again and (re)affecting the same DOM elements. Is it the case that, on next firing, the body is thrown out and re-inserted, so any DOM changes are happening on "new" elements?